queue/driver_mongo.go

```diff
@@ -774,6 +774,11 @@ RETRY:
 					continue CURSOR
 				}
 
+				if d.scopesInUse(ctx, job.Scopes()) {
```

Comment:
if the job requires scopes (output of job.Scopes()), and any of those scopes are currently set, skip this job and move on. 

This saves time because doing a count of keys in a small index is quicker than attempting to do a replaceOne on a document only to have it fail because of a duplicate key exception.